src/lib/tools.ts

```diff
@@ -0,0 +1,245 @@
+
+export const tunnelAsync = Bluebird.promisify(tunnel);
+
+export async function getDirectories(path: string): Promise<string[]> {
+	const directories = (await fs.readdir(path)) as string[];
+	return directories.filter(async (name: string) => {
+		const itemPath = join(path, name);
+		const itemStat = await fs.stat(itemPath);
+		return itemStat.isDirectory();
+	});
+}
+
+/**
+ * Returns an absolute path for a path, when in basePath
+ * @param {string} path: The file path, (or an absolute path)
+ * @param {string} basePath: The base path
+ * @returns {string} An asolute path
+ */
+export function getAbsolutePath(path: string, basePath: string): string {
+	return isAbsolute(path) ? path : join(basePath, path);
+}
+
+export function getAbsoluteURI(uri: string, basePath: string): string {
+	if (gitURI(uri)) {
+		throw new NotImplementedError('Git URI support not implemented yet');
+	} else if (localPathURI(uri)) {
+		return getAbsolutePath(uri, basePath);
+	} else {
+		throw new UnsupportedError('URI type not supported yet');
+	}
+}
+
+export async function loadFromFile(
+	filePath: string,
+	errorMessage: string = '',
+): Promise<any> {
+	try {
+		const buffer = await fs.readFile(filePath);
+		return yamljs.parse(buffer.toString('utf8'));
+	} catch (e) {
+		throw new FileLoadError(errorMessage + e.message);
+	}
+}
+
+export function gitURI(uri: string): boolean {
+	return /((git|ssh|http|https)|(git@[\w\.]+))(:(\/\/)?)([\w\.@\:/\-]+)/.test(
+		uri,
+	);
+}
+
+export function localPathURI(uri: string): boolean {
+	return /^([a-zA-Z0-9_/\-.])+$/.test(uri);
+}
+
+export async function loadFromURI(
+	URI: string,
+	path: string,
+	errorMessage?: string,
+): Promise<any> {
+	// TODO: support git URI
+	if (gitURI(URI)) {
+		throw new UnsupportedError('Git URI support not implemented yet');
+	}
+	try {
+		if (localPathURI(URI)) {
+			return await loadFromFile(join(URI, path), errorMessage);
+		}
+	} catch (e) {
+		throw new URILoadError(`Error loading ${path} from ${URI}\n${e.message}`);
+	}
+}
+
+export function getBasePath(path: string): string {
+	return dirname(resolve(path));
+}
+
+export function convertRelativePaths({
+	conf,
+	basePath,
+}: {
+	conf: any;
+	basePath: string;
+}): any {
+	// Convert relative to absolute URIs
+	const keys = [
+		'productRepo',
+		'archiveStore',
+		'encryptionKeyPath',
+		'envFile.path',
+		'yamlFile.path',
+		'kubernetes.kubeConfigPath',
+		'kubernetes.bastion.key',
+		'compose.socket',
+	];
+
+	for (const k of keys) {
+		const value = _.get(conf, k);
+		if (value) {
+			_.set(conf, k, getAbsoluteURI(value, basePath));
+		}
+	}
+	return conf;
+}
+
+export async function readFromURI(
+	URI: string,
+	path: string,
+	cachePath?: string,
+): Promise<any> {
+	// TODO: support git URI
+	if (gitURI(URI)) {
+		throw new UnsupportedError('Git URI support not implemented yet');
+	} else if (localPathURI(URI)) {
+		return (await fs.readFile(join(URI, path))).toString('utf8');
```

Comment:
Why not use `loadFromFile` which will do the conversion for you (given it's been defined)?

Reply:
This function should return a string `readFromUri. `loadFromFile` loads a yaml file, and returns the parsed object.

Reply:
`readFromUri`, then :)